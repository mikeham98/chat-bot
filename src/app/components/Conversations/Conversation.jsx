import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from '../../../themes/components/conversations.scss';

export default class Conversation extends React.PureComponent {
    returnClassNames() {
        const {selected, unread} = this.props;
        const classNames = [styles.conversationWrapper];
        if (selected) {
            classNames.push(styles.conversationSelected);
        }
        if (unread) {
            classNames.push(styles.conversationUnread);
        }
        return classnames(classNames);
    }

    render() {
        const {onClick, image, name, previewMessage} = this.props;
        return (
            <div onClick={onClick} className={this.returnClassNames()}>
                <img src={image}/>
                <div className={styles.conversationPreview}>
                    <span className={styles.conversationPreviewName}>{name}</span>
                    {previewMessage &&
                    <span className={styles.conversationPreviewMessage}>{previewMessage}</span>}
                </div>
            </div>
        );
    }
};

Conversation.defaultProps = {
    previewMessage: ''
};

Conversation.propTypes = {
    selected: PropTypes.bool,
    unread: PropTypes.bool,
    onClick: PropTypes.func,
    previewMessage: PropTypes.string,
    image: PropTypes.string,
    name: PropTypes.string,
};