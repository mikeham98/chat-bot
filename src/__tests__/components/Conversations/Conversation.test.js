import React from 'react';
import {shallow} from 'enzyme';
import Conversation from '../../../app/components/Conversations/Conversation'

describe('Conversation', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<Conversation {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            selected: false,
            image: 'http://www.image.co.uk/img1',
            name: 'John Smith'
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a div with an image and a name', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a div with a className of conversationWrapper and conversationSelected', () => {
            props.selected = true;
            expect(wrapper()).toMatchSnapshot();
        });
    });
});
