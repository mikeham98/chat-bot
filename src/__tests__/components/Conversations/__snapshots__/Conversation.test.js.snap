// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conversation render should return a div with a className of conversationWrapper and conversationSelected 1`] = `
<div
  className="conversationWrapper conversationSelected"
>
  <img
    src="http://www.image.co.uk/img1"
  />
  <span>
    John Smith
  </span>
</div>
`;

exports[`Conversation render should return a div with an image and a name 1`] = `
<div
  className="conversationWrapper"
>
  <img
    src="http://www.image.co.uk/img1"
  />
  <span>
    John Smith
  </span>
</div>
`;
